vault/plugin_catalog.go

```diff
@@ -269,10 +539,10 @@ func (c *PluginCatalog) Set(ctx context.Context, name string, pluginType consts.
 	c.lock.Lock()
 	defer c.lock.Unlock()
 
-	return c.setInternal(ctx, name, pluginType, command, args, env, sha256)
+	return c.setInternal(ctx, name, pluginType, false, command, args, env, sha256)
```

Comment:
How come we're hardcoding `multiplexingSupport` to false in here? How would this work if a database plugin that supports multiplexing was registered to the catalog manually (i.e. not by the auto-registration from dev mode which I think goes through `UpgradePlugins`)?

Reply:
Nevermind! The catalog registration process in dev mode goes through a normal backend request for registering a plugin, so the codepath hits this (and we've seen mux'ed plugins registered this way working fine). It's a bit odd that we're returning false even though things work, so it might be good to add a comment here on why this is okay.

Reply:
**Note**: You will need to expand `plugin_catalog.go` in GitHub for any of the following links to work.

This is just the initial value that doesn't actually affect anything during registration. Actually, I was looking at this today and considered removing this parameter and hardcoding false [here](https://github.com/hashicorp/vault/pull/14033/files#diff-3a01767027f547971b29faa241c1097525a3a5a7f3ae9d747877117238226cc5R573). Or it can probably just be omitted entirely. 

On plugin registration, we just need a pluginRunner. The multiplexingSupport field will be set to false for both muxed and non-muxed plugins. We don't actually know if a plugin supports multiplexing until we run the plugin and ask the client via [pluginutil.MultiplexingSupported](https://github.com/hashicorp/vault/pull/14033/files#diff-3a01767027f547971b29faa241c1097525a3a5a7f3ae9d747877117238226cc5R370). Once that call tells us whether multiplexing is supported we set it on the pluginRunner [here](https://github.com/hashicorp/vault/pull/14033/files#diff-3a01767027f547971b29faa241c1097525a3a5a7f3ae9d747877117238226cc5R593) and write it to storage. Then when NewPluginClient is called when writing a database config, we call [PluginCatalog.get](https://github.com/hashicorp/vault/pull/14033/files#diff-3a01767027f547971b29faa241c1097525a3a5a7f3ae9d747877117238226cc5R211) to get the pluginRunner that is now able to tell us if the plugin is multiplexed.

When `NewPluginClient` is called during plugin registration, we don't care if it is multiplexed or not because we will always want to spawn a process. When `NewPluginClient` is called during config write, we will know if it is multiplexed.